CGenerator/src/main/java/org/example/IFeature.java

```diff
@@ -1,6 +1,77 @@
 package org.example;
 
+
+/*
+    Interface containing all the functions that must be implemented to make a useful
+    feature-class.
+
+    Read the note on the constructor carefully.
+ */
+
 public interface IFeature {
+    /*
+       It is not possible to force a constructor declaration using an interface. However,
+       every IFeature-based class is supposed to implement a constructor like this:
+
+            final CGenerator generator;
+            public [class name](CGenerator generator){
+                this.generator = generator;
+            }
+
+       This will make sure that every feature knows its calling object, which it can use to
+       request c-code. For example: this feature's code may want to have a random struct. If it
+       does so, it will ask the generator to pick one for him. The generator will then give
+       one from its database, or have a feature (possibly even the same) create one.
+
+     */
+
+    /*
+       It is not possible to force a constructor declaration using an interface. However,
+       every IFeature-based class is supposed to implement a constructor like this:
+
+            final CGenerator generator;
+            public [class name](CGenerator generator){
+                this.generator = generator;
+            }
+
+       This will make sure that every feature knows its calling object, which it can use to
+       request c-code. For example: this feature's code may want to have a random struct. If it
+       does so, it will ask the generator to pick one for him. The generator will then give
+       one from its database, or have a feature (possibly even the same) create one.
+
+     */
+
+
+    /*
+       It is not possible to force a constructor declaration using an interface. However,
+       every IFeature-based class is supposed to implement a constructor like this:
+
+            final CGenerator generator;
+            public [class name](CGenerator generator){
+                this.generator = generator;
+            }
+
+       This will make sure that every feature knows its calling object, which it can use to
+       request c-code. For example: this feature's code may want to have a random struct. If it
+       does so, it will ask the generator to pick one for him. The generator will then give
+       one from its database, or have a feature (possibly even the same) create one.
+
+     */
```

Comment:
Drie keer hetzelfde?

Reply:
Dat is om ervoor te zorgen dat niemand het over het hoofd ziet :-)
(En het is het resultaat van (drie)dubbel rebasen, vermoed ik...)

Maar goed: ik zal vanavond voor zorgen dat dit iets-te-veel-van-het-goede eruit gaat.